### PR TITLE
Add missing pieces for URI variables

### LINKED ISSUE #53: Thing::uri_variables

### PR DESCRIPTION
**Pull request by dodomorandi**
Now we should be fine. Please, make a double check because it took me more time to dig around what was missing than the writing the actual changes.

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -438,6 +441,19 @@ impl<Other: ExtendableThing, Status> ThingBuilder<Other, Status> {
             }
         };
 
+        let invalid_uri_variables = uri_variables
```
Comment (lu-zero) [inline src/builder.rs:444]: `InteractionAffordance::uri_variables` needs a similar check. maybe factor it out to a function.

File: src/builder.rs [hunk 2]
```diff
@@ -3566,7 +3582,7 @@ mod tests {
                     .ext(())
                     .ext(DataSchemaExtC { t: 5 })
                     .finish_extend()
-                    .array()
```
[the remark this answers is not in the document]
Comment (dodomorandi) [reply src/builder.rs:3569]: It is, the new check does not allow to use an array (or an object) for uri_variables. The test fails if I don't make this change.

File: src/builder.rs (hunk 1 above)
Comment (dodomorandi) [reply src/builder.rs:444]: Good catch, I did not realize it.

**Comment by dodomorandi** [comment]
Once things are reviewed, I will rebase+fixup the last commit (just to ease to review with a meaningful diff)

**Review by lu-zero — APPROVED** [review]
land it anytime :)

**Pull request merged**